### DISCUSSION TITLE
How to create a rectangle Waveform?

### DISCUSSION THREAD

**Discussion by icegh** [ar1st0crat/NWaves#69] [Q&A] [answered]
[dear community,

i am new in c# and also using first time NWaves lib.

Can anybody helps me how to create a rectangle like that (i have attached a image).
i have no idead which type of signal builder i should use. has already tried all but no chance to get the same results.

In audacity i can simple create this via menu. But how i can do that with the NWaves lib?
![14-11-_2022_20-33-34](https://user-images.githubusercontent.com/100025138/201749613-cdf4fb63-7139-4bbb-a322-747ad57da97a.jpg)

thank you so much]([url](url))

**Comment by ar1st0crat** [comment, marked as answer]
Hi, does this code generate the waveform you need?

```C#

var samplingFrequency = 44100/*Hz*/;

DiscreteSignal signal = 
    new SquareWaveBuilder()
        .SetParameter("frequency", 432.0/*Hz*/)
        .SetParameter("min", -0.8)
        .SetParameter("max", 0.8)
        .OfDuration(4*60 + 16) // seconds
        .SampledAt(samplingFrequency)
        .Build();

```

**Reply by icegh** [reply]
Thank you so much, thats solved my problem.
